lib/api/core/models/repositories/repository.js

```diff
@@ -1,344 +1,421 @@
+   * @param {object} [options] - Optional options for the current operation
+   * @param {string} [collection]
+   * @returns {Promise}
+   */
+  persistToCache (object, options = {}, collection = this.collection) {
+    const ttl = options.ttl !== undefined ? options.ttl : this.ttl;
+    let key;
+
+    if (collection === null) {
+      return Promise.reject(new InternalError('The collection must be specified.'));
+    }
+
+    key = options.key || this.getCacheKey(object._id, collection);
+
+    if (ttl === false) {
+      return this.cacheEngine.set(key, JSON.stringify(this.serializeToCache(object)))
+        .then(() => object);
+    }
+
+    return this.cacheEngine.volatileSet(key, JSON.stringify(this.serializeToCache(object)), ttl)
       .then(() => object);
   }
 
-  return this.cacheEngine.volatileSet(key, JSON.stringify(this.serializeToCache(object)), ttl)
-    .then(() => object);
-};
+  /**
+   * Removes the object from the Cache Engine
+   * The opts optional parameters currently accepts only 1 option:
+   *   key: if provided, removes the given key instead of the default one (<collection>/<id>)
+   *
+   * @param {string} id
+   * @param {object} [options] - optional options for the current operation
+   * @param {string} [collection]
+   * @returns {Promise}
+   */
+  deleteFromCache (id, options = {}, collection = this.collection) {
+    let key;
+
+    if (collection === null) {
+      return Promise.reject(new InternalError('The collection must be specified.'));
+    }
 
-/**
- * Removes the object from the Cache Engine
- * The opts optional parameters currently accepts only 1 option:
- *   key: if provided, removes the given key instead of the default one (<collection>/<id>)
- *
- * @param {string} id
- * @param {object} [opts={}] optional options for the current operation
- * @returns {Promise}
- */
-Repository.prototype.deleteFromCache = function repositoryDeleteFromCache (id, opts) {
-  var
-    options = opts || {},
-    key = options.key || this.cacheKeyPrefix + id;
-
-  return this.cacheEngine.remove(key);
-};
-
-Repository.prototype.refreshCacheTTL = function repositoryRefreshCacheTTL (object, opts) {
-  var
-    options = opts || {},
-    key = options.key || this.cacheKeyPrefix + object._id,
-    ttl = this.ttl;
-
-  if (options.ttl !== undefined) {
-    ttl = options.ttl;
+    key = options.key || this.getCacheKey(id, collection);
+
+    return this.cacheEngine.remove(key);
   }
 
-  if (ttl === false) {
-    return this.cacheEngine.persist(key);
+  /**
+   * @param {object} object
+   * @param {object} [options] - optional options for the current operation
+   * @param {string} [collection]
+   * @returns {*}
+   */
+  refreshCacheTTL (object, options = {}, collection = this.collection) {
+    const ttl = options.ttl !== undefined ? options.ttl : this.ttl;
+    let key;
+
+    if (collection === null) {
+      return Promise.reject(new InternalError('The collection must be specified.'));
```

Comment:
It can not happen due to a bad request, it might come from an bad plugin implementation, will use this one instead.